apps/dashboard/src/components/conditions-editor/conditions-editor.tsx

```diff
@@ -58,27 +76,119 @@ function InternalConditionsEditor({
   query,
   onQueryChange,
   saveForm,
+  enhancedVariables,
 }: {
-  fields: Field[];
+  fields: Field[] | EnhancedField[];
   variables: LiquidVariable[];
   isAllowedVariable: IsAllowedVariable;
   query: RuleGroupType;
   onQueryChange: (query: RuleGroupType) => void;
   saveForm: () => void;
+  enhancedVariables?: EnhancedLiquidVariable[];
 }) {
   const context = useMemo(() => ({ variables, isAllowedVariable, saveForm }), [variables, isAllowedVariable, saveForm]);
 
+  // Create field data map for enhanced field lookups
+  const fieldDataMap = useMemo(() => {
+    if (!enhancedVariables) return new Map();
+
+    return new Map(
+      enhancedVariables.map((variable) => [
+        variable.name,
+        {
+          name: variable.name,
+          label: variable.displayLabel || variable.name,
+          value: variable.name,
+          dataType: variable.dataType,
+          inputType: variable.inputType,
+          format: variable.format,
+        } as EnhancedField,
```

Comment:
as below, do you remember why this casting is needed?
consider converting to `satisfies` to validate the type.
```suggestion
        } satisfies EnhancedField,
```